data/pref.xml

```diff
@@ -46,6 +46,10 @@
       <value id="ACTIVE_LAYER" value="0" />
       <value id="ALL_LAYERS" value="1" />
     </enum>
+    <enum id="PixelConnectivity">
+      <value id="EIGHT_CONNECTED" value="0" />
+      <value id="FOUR_CONNECTED" value="1" />
```

Comment:
The only thing I would change, is the order of these constants, I would put `FOUR_CONNECTED` = 0, and `EIGHT_CONNECTED` = 1 (so 0 is the default value). Anyway another change I would make (but I can do it later) is to move this enum to the same place as in `floodfill()` function, so the function can receive the `enum` value instead of a `isEightConnected` boolean value.

Reply:
I changed the order of the constants as requested, but I couldn't figure out where I needed to move the enum to -- I'd be more than happy to refactor the bool to enum myself if it's easier for you.  I just need a pointer to where to get started :)